src/views/workflow-history/hooks/use-event-expansion-toggle.ts

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useState } from 'react';
+
+import omit from 'lodash/omit';
+
+import { type HistoryEvent } from '@/__generated__/proto-ts/uber/cadence/api/v1/HistoryEvent';
+
+import {
+  type EventCollapseState,
+  type GetIsEventExpanded,
+  type ToggleIsEventExpanded,
+  type ToggleIsExpandAllEvents,
+} from './use-event-expansion-toggle.types';
+
+export default function useEventExpansionToggle({
```

Comment:
nit
```suggestion
export default function useEventExpansionState({
```

Reply:
useToggle is a common naming convention to describe toggle-able states. So would leave it for better behavior description